apps/parsley/src/hooks/useSections/utils.ts

```diff
@@ -150,5 +153,76 @@ const parseSections = (logs: string[]): SectionData => {
   return { commands, functions };
 };
 
-export { parseSections, reduceFn };
+/**
+ * getSectionStateWithOpenSectionBasedOnLineNumber Gets the next section state where all sections that contain the lineNumber values(s)
+ * are open. The return value returns a boolean representing if the next state is different from current state
+ * as well as the next section state.
+ * @param props is an object with a lineNumber key(s), sectionData and sectionState.
+ * @param props.sectionData is the parsed section data
+ * @param props.sectionState is the current section state
+ * @param props.lineNumber is a number or an array of numbers that represent raw log line numbers.
+ * @returns [boolean, SectionState]
+ */
+const getSectionStateWithOpenSectionBasedOnLineNumber = ({
+  lineNumber,
+  sectionData,
+  sectionState,
+}: {
+  sectionData: SectionData;
+  sectionState: SectionState;
+  lineNumber: number | number[];
+}): [boolean, SectionState] => {
+  const lineNumberArray = conditionalToArray(lineNumber, true);
+  const sectionContainingLine = sectionData.commands.filter((section) =>
+    lineNumberArray.some((n) => includesLineNumber(section, n)),
+  );
```

Comment:
is it possible for this function to just accept a `number[]` and do the conversion in the parent function?
```suggestion
  lineNumbers: number[];
}): [boolean, SectionState] => {
  const sectionContainingLine = sectionData.commands.filter((section) =>
    lineNumbers.some((n) => includesLineNumber(section, n)),
  );
```

Reply:
I pushed the conversion all the way up into the hook in a setting function separate from the `useState`